core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -297,4 +298,16 @@ private static Schema copyRecord(Schema record, List<Schema.Field> newFields) {
   private static boolean isOptionSchemaWithNonNullFirstOption(Schema schema) {
     return AvroSchemaUtil.isOptionSchema(schema) && schema.getTypes().get(0).getType() != Schema.Type.NULL;
   }
+
+  private static Schema copyUnion(Schema record, List<Schema> visitResults) {
+    List<Schema> alts = Lists.newArrayListWithExpectedSize(visitResults.size());
+    for (int i = 0; i < visitResults.size(); i++) {
+      if (visitResults.get(0) == null) {
```

Comment:
Why do we use `visitResults.get(0) == null` instead of `visitResults.get(i) == null`?
And could you add comment for this method, preferably with an example.

Reply:
Good find, this is a typo, it should be `get (i)`. This method is more of implementation detail, the reason is for primitive types, the `visitResult` will be null, so we want to reuse the primitive types from the original union schema, while for nested types, we want to use the `visitResult` because they have content from the previous recursive calls.